addons/binding/org.openhab.binding.nikohomecontrol/src/main/java/org/openhab/binding/nikohomecontrol/handler/NikoHomeControlBridgeHandler.java

```diff
@@ -0,0 +1,238 @@
+import org.eclipse.smarthome.core.types.Command;
+import org.openhab.binding.nikohomecontrol.internal.discovery.NikoHomeControlDiscoveryService;
+import org.openhab.binding.nikohomecontrol.internal.protocol.NikoHomeControlCommunication;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+
+/**
+ * {@link NikoHomeControlBridgeHandler} is the handler for a Niko Home Control Gateway and connects it to
+ * the framework.
+ *
+ * @author Mark Herwege
+ */
+public class NikoHomeControlBridgeHandler extends BaseBridgeHandler {
+
+    private Logger logger = LoggerFactory.getLogger(NikoHomeControlBridgeHandler.class);
+
+    private NikoHomeControlCommunication nhcComm;
+
+    private ScheduledFuture<?> refreshTimer;
+
+    private InetAddress addr;
+    private Integer port;
+
+    private NikoHomeControlDiscoveryService nhcDiscovery;
+
+    public NikoHomeControlBridgeHandler(Bridge nikoHomeControlBridge) {
+        super(nikoHomeControlBridge);
+    }
+
+    @Override
+    public void handleCommand(ChannelUID channelUID, Command command) {
+        // There is nothing to handle in the bridge handler
+    }
+
+    @Override
+    public void initialize() {
+
+        logger.debug("Niko Home Control: initializing bridge handler");
+
+        Configuration config = this.getConfig();
+
+        port = ((Number) config.get(CONFIG_PORT)).intValue();
+        logger.debug("Niko Home Control: bridge handler port {}", port);
+
+        try {
+            // If hostname or address was provided in the configuration, try to use this to for bridge and give error
+            // when hostname parameter was not valid.
+            // No hostname provided is a valid configuration, therefore allow null addr to pass through.
+            if (config.get(CONFIG_HOST_NAME) != null) {
+                logger.debug("Niko Home Control: bridge handler host {}", config.get(CONFIG_HOST_NAME));
+                addr = InetAddress.getByName((String) config.get(CONFIG_HOST_NAME));
+            }
+        } catch (UnknownHostException e) {
+            updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.OFFLINE.COMMUNICATION_ERROR,
+                    "Niko Home Control: cannot resolve bridge IP with hostname " + config.get(CONFIG_HOST_NAME));
+            return;
+        }
+
+        Integer refreshInterval = ((Number) config.get(CONFIG_REFRESH)).intValue();
+
+        createCommunicationObject();
+        setupRefreshTimer(refreshInterval);
+
+    }
+
+    /**
+     * Create communication object to Niko Home Control gateway and start communication.
+     * Trigger discovery when communication setup is successful.
+     *
+     * @param nhcAddr IP address or subnet for gateway
+     * @param port
+     */
+    private void createCommunicationObject() {
+
+        scheduler.submit(new Runnable() {
+
+            @Override
+            public void run() {
+                try {
+                    for (int i = 0; i < 3; i++) {
```

Comment:
I tried building in some resilience to a bad connection, but in my setup it does connect from the first pass, or not at all. So I removed it.